packages/ovm-truffle-provider-wrapper/index.ts

```diff
@@ -1,4 +1,39 @@
-module.exports = (provider: any) => {
+import { execSync, spawn } from 'child_process'
+
+/**
+ * Starts a local OVM node process for testing. It will be killed when the current process terminates.
+ */
+const startLocalNode = () => {
+  const runText = `(async function(){ const {runFullnode} = require('@eth-optimism/rollup-full-node');runFullnode();})();`
+
+  // Assumes this process was kicked off with node, but that's true for `truffle test` and `yarn ...`
+  const sub = spawn(process.argv[0], [`-e`, `${runText}`])
+
+  sub.on('error', (e) => {
+    // tslint:disable-next-line:no-console
+    console.error(
+      `Local server could not be started. Error details: ${e.message}, Stack: ${e.stack}`
+    )
+  })
+
+  // This reliably stops the local server when the current process exits.
+  process.on('exit', () => {
+    try {
+      sub.kill()
+    } catch (e) {
+      /*swallow any errors */
+    }
+  })
+
+  // TODO: This is hacky. If host / port become configurable, spawn a node process to ping it or something better.
+  execSync(`sleep 3`)
```

Comment:
Can we `execSync` directly on `runText` or does that break error handling/create zombie processes? I think we'd also need to add an `await` before `runFullnode` on line 7

I've been thinking and thinking if there's a better way to do this but I think our hands are tied 😕

Reply:
`runFullnode` runs a server, so its promise won't be resolved until the server is stopped. For the same reason, we can't run sync because the point is we want the server running after the call.